src/coffee/sync/utils/product.coffee

```diff
@@ -248,15 +247,15 @@ class ProductUtils extends BaseUtils
         index = key.substring(1)
       if index
         delete price.discounted # we don't need this for mapping the action
-
         # if id is unchanged, we initiate a changePrice action with priceId
         if not _.isArray(price) and not price.id and old_variant.prices.length > index
-          changeAction = @_buildChangePriceActionWithPriceId(new_variant, index)
+          changeAction = @_buildChangePriceActionWithPriceId(old_variant, new_variant, index)
```

Comment:
`changePrice` actions always require price id so maybe you can skip those suffixes and rename to -> `_buildChangePriceAction`?

Also if we match by scope why do we still check for existence of new price `id` in line 251?

Reply:
In the case that a price scope is to be changed, it will be matched by ID